kore/src/Kore/Step/Axiom/Evaluate.hs

```diff
@@ -22,6 +22,9 @@ import Kore.Internal.TermLike
     ( TermLike
     )
 import qualified Kore.Internal.TermLike as TermLike
+import qualified Kore.Predicate.Predicate as Syntax
```

Comment:
Well, I was a bit conflicted about that... The main issue was something like this: the substitution is useful when we want to actually substitute something or we want to merge it with another substitution, or something. Sending it separately was suggesting that it was going to be useful, but I actually wanted to send an opaque predicate which was going to be used only as an additional condition by the SMT (of course, it can include the substitution). Maybe I should have also wrapped it in a newtype to make that even more clear.

On the other hand, yes, this is breaking the current convention, so it feels weird. Should I make it an `Internal.Predicate`?

Reply:
You're right, it is awkward. In fact, we never send a substitution to the solver anyway because we abort if it is not normalized. So, I think keep it the way you have it.